cv_next/app/cv/[cvId]/components/commentSection/comment.tsx

```diff
@@ -281,62 +298,32 @@ export default function Comment({
       commentToAdd.parent_comment_Id = comment.parent_comment_Id;
     }
 
-    const optimisticData = (
-      prevData: CommentModel[] | undefined
-    ): CommentModel[] => {
-      const newCommentsOfComments = new Map<string, CommentModel[]>(
-        prevData?.reduce((acc, cur) => {
-          // Use 'root' for top-level comments
-          const key = cur.parent_comment_Id || "root";
-          if (!acc.has(key)) {
-            acc.set(key, []);
-          }
-          acc.get(key)!.push(cur);
-          return acc;
-        }, new Map<string, CommentModel[]>()) || []
-      );
-
-      const parentId = comment.id;
-      if (!newCommentsOfComments.has(parentId)) {
-        newCommentsOfComments.set(parentId, []);
-      }
-      newCommentsOfComments
-        .get(parentId)!
-        .push(commentToAdd as unknown as CommentModel);
-
-      return Array.from(newCommentsOfComments.values()).flat();
-    };
+    mutate(comment.document_id, (currentSubComments: CommenterModel[] = []) => [
+      ...currentSubComments,
+      {
+        ...commentToAdd,
+        id: Date.now().toString(),
```

Comment:
Also- what? Why define id as date.now()? Why not let the db create the id?

Reply:
It's a temporary thing until we load the actual comment that we get from the server(which is done immediately after although we don't trigger re-render this way)